### PR TITLE
Drafts Search and compare section for NLP docs

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR adds the "Search and compare" page to the NLP docs.
The page contains the conceptual docs for the Text embedding task.

### Preview

[Search and compare](https://stack-docs_1914.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-search-compare.html)

### CODE REVIEW

**Review by joshdevins — CHANGES_REQUESTED** [review]

**Review by joshdevins — APPROVED** [review]

File: docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Text embedding is a task which produces a mathematical representation of text 
+called an embedding. The {ml} model turns the text into an array of numerical 
+values (a vector). Pieces of content with similar meaning have similar 
+representation. This means it is possible to determine whether different pieces 
+of text are either semantically similar, different, or even opposite by using a 
+mathematical similarity function.
+
+This task is only responsible for producing the embedding alone. When the 
+embedding is created, it can be stored in a `dense_vector` field and used at 
+search time. Then you can apply the similarity function to an embedding
```
Comment (lcawl) [inline docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc:22]: > Then you can apply the similarity function to an embedding representing a query and create semantic search capability.

This sentence is not clear to me. Perhaps this is better?

> For example, you can use these vectors in a k-nearest neighbor (kNN) search to achieve semantic search capabilities.

File: docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+compare difference pieces of text.
+
+
+[discrete]
+[[ml-nlp-text-embedding]]
+== Text embedding
+
+Text embedding is a task which produces a mathematical representation of text 
+called an embedding. The {ml} model turns the text into an array of numerical 
+values (a vector). Pieces of content with similar meaning have similar
```
[the remark this answers is not in the document]
Comment (joshdevins) [reply docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc]: An embedding is a kind of (dense) vector. When we use the term embedding, we are referring to a (dense) vector that has certain properties. In the case of embeddings created from a text similarity model, the vectors have the properties that when they are close together in their high-dimensional space (each dimension of the vector is a dimension of a "space", if there are 3-dimensions, it's a 3D space), they have similar semantic meaning. The "closeness" is measured using a metric like dot-product, cosine similarity, etc. So a smaller cosine similarity between query embedding `Q` and an indexed embedding `A` means that they are semantically more similar than perhaps another indexed embedding `B`. Does that makes sense?

File: docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc (hunk 1 above)
Comment (joshdevins) [reply docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc]: We should link out to the new kNN docs: https://www.elastic.co/guide/en/elasticsearch/reference/master/knn-search.html

**Comment by joshdevins** [comment]
I added a few minor comments, and some context. I think the only thing missing is a link out to the new kNN search endpoint.

File: docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+values (also known as a _vector_). Pieces of content with similar meaning have 
+similar representations. This means it is possible to determine whether 
+different pieces of text are either semantically similar, different, or even 
+opposite by using a mathematical similarity function.
+
+This task is responsible for producing only the embedding. When the 
+embedding is created, it can be stored in a 
+{ref}/dense-vector.html[dense_vector] field and used at search time. For 
+example, you can use these vectors in a 
+{ref}/knn-search.html[k-nearest neighbour (kNN) search] to achieve semantic
```
Comment (szabosteve) [inline docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc:25]: @joshdevins The link that points to the kNN search docs is here in line 25.

File: docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/en/stack/ml/nlp/ml-nlp-search-compare.asciidoc]: I've added the link via e3b2698.

**Review by lcawl — APPROVED** [review]
LGTM!

**Pull request merged**